plan/plans/show.go

```diff
@@ -676,3 +685,39 @@ func (s *ShowPlan) fetchShowTriggers(ctx context.Context) error {
 func (s *ShowPlan) fetchShowProcedureStatus(ctx context.Context) error {
 	return nil
 }
+
+func (s *ShowPlan) fetchShowIndex(ctx context.Context) error {
+	tb, err := s.getTable(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	for _, idx := range tb.Indices() {
+		for i, col := range idx.Columns {
+			nonUniq := 1
+			if idx.Unique {
+				nonUniq = 0
+			}
+			var subPart interface{}
+			if col.Length != -1 {
```

Comment:
When you do not specify index column prefix length, it  will be -1.
https://github.com/pingcap/tidb/blob/master/parser/parser.y#L4099